### ISSUE TITLE
[Legal] Compiled JavaScript code don't include WebCord's license.

### ISSUE THREAD

**Issue by SpacingBat3** [smartfrigde/armcord#163]
### TL;DR
ARMCord uses parts of WebCord's code and does not respect WebCord's software license. Please move it outside of source files, preferably include alongisde the ARMCord's license with note similar to the one below:
> Parts of WebCord's code used in ARMCord are licensed under MIT license:
>
> Copyright (c) 2020-2022 Dawid Papiewski "SpacingBat3"
>
> Permission is hereby granted, free of charge, to any person obtaining a copy
> of this software and associated documentation files (the "Software"), to deal
> in the Software without restriction, including without limitation the rights
> to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
> copies of the Software, and to permit persons to whom the Software is
> furnished to do so, subject to the following conditions:
>
> The above copyright notice and this permission notice shall be included in all
> copies or substantial portions of the Software.

As long as this part of the license is distributed with **both** ARMCord's code and compiled code, it won't violate the WebCord's license conditions.

### Description

ARMCord uses some parts of WebCord's project like [`socket.ts`] to implement its own WebSocket Server functionality. However WebCord's code is copyrighted under the terms of MIT License and ARMCode does not include it in all essential files that are part of WebCord. For an example, it is not present in `socket.js` as it can be seen below:

<details>
<summary><code>socket.js</code> from <a href="https://github.com/ArmCord/ArmCord/releases/tag/v3.0.7"><b>ARMCord 3.0.7</b></a></summary>

```js
"use strict";
// MIT License
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const window_1 = require("./window");
async function wsLog(message, ...args) {
    console.log("[WebSocket]" + message, ...args);
}
/** Generates an inclusive range (as `Array`) from `start` to `end`. */
function range(start, end) {
    return Array.from({ length: end - start + 1 }, (_v, k) => start + k);
}
function isInviteResponse(data) {
    if (!(data instanceof Object))
        return false;
    if (data?.cmd !== "INVITE_BROWSER")
        return false;
    if (typeof data?.args?.code !== "string")
        return false;
    if (typeof data?.nonce !== "string")
        return false;
    return true;
}
const messages = {
    /**
     * A fake, hard-coded Discord command to spoof the presence of
     * official Discord client (which makes browser to actually start a
     * communication with the ArmCord).
     */
    handShake: {
        /** Message command. */
        cmd: "DISPATCH",
        /** Message data. */
        data: {
            /** Message scheme version. */
            v: 1,
            /** Client properties. */
            config: {
                /** Discord CDN host (hard-coded for `dicscord.com` instance). */
                cdn_host: "cdn.discordapp.com",
                /** API endpoint (hard-coded for `dicscord.com` instance). */
                api_endpoint: "//discord.com/api",
                /** Client type. Can be (probably) `production` or `canary`. */
                environment: "production"
            }
        },
        evt: "READY",
        nonce: null
    }
};
/**
 * Tries to reserve the server at given port.
 *
 * @returns `Promise`, which always resolves (either to `Server<WebSocket>` on
 *          success or `null` on failure).
 */
async function getServer(port) {
    const { WebSocketServer } = await Promise.resolve().then(() => __importStar(require("ws")));
    return new Promise((resolve) => {
        const wss = new WebSocketServer({ host: "127.0.0.1", port });
        wss.once("listening", () => resolve(wss));
        wss.once("error", () => resolve(null));
    });
}
/**
 * Tries to start a WebSocket server at given port range. If it suceed, it will
 * listen to the browser requests which are meant to be sent to official
 * Discord client.
 *
 * Currently it supports only the invitation link requests.
 *
 */
async function startServer() {
    function isJsonSyntaxCorrect(string) {
        try {
            JSON.parse(string);
        }
        catch {
            return false;
        }
        return true;
    }
    /** Known Discord instances, including the official ones. */
    const knownInstancesList = [
        ["Discord", new URL("https://discord.com/app")],
        ["Discord Canary", new URL("https://canary.discord.com/app")],
        ["Discord PTB", new URL("https://ptb.discord.com/app")],
        ["Fosscord", new URL("https://dev.fosscord.com/app")]
    ];
    let wss = null, wsPort = 6463;
    for (const port of range(6463, 6472)) {
        wss = await getServer(port);
        if (wss !== null) {
            void wsLog("ArmCord is listening at " + port.toString());
            wsPort = port;
            break;
        }
    }
    if (wss === null)
        return;
    let lock = false;
    wss.on("connection", (wss, request) => {
        const origin = request.headers.origin ?? "https://discord.com";
        let known = false;
        for (const instance of knownInstancesList) {
            if (instance[1].origin === origin)
                known = true;
        }
        if (!known)
            return;
        wss.send(JSON.stringify(messages.handShake));
        wss.once("message", (data, isBinary) => {
            if (lock)
                return;
            lock = true;
            let parsedData = data;
            if (!isBinary)
                parsedData = data.toString();
            if (isJsonSyntaxCorrect(parsedData))
                parsedData = JSON.parse(parsedData);
            if (isInviteResponse(parsedData)) {
                // Replies to browser, so it finds the communication successful.
                wss.send(JSON.stringify({
                    cmd: parsedData.cmd,
                    data: {
                        invite: null,
                        code: parsedData.args.code
                    },
                    evt: null,
                    nonce: parsedData.nonce
                }));
                (0, window_1.createInviteWindow)();
                const child = window_1.inviteWindow;
                if (child === undefined)
                    return;
                void child.loadURL(origin + "/invite/" + parsedData.args.code);
                child.webContents.once("did-finish-load", () => {
                    child.show();
                });
                child.webContents.once("will-navigate", () => {
                    lock = false;
                    child.close();
                });
                child.on("close", (e) => {
                    lock = false;
                });
                // Blocks requests to ArmCord's WS, to prevent loops.
                child.webContents.session.webRequest.onBeforeRequest({
                    urls: ["ws://127.0.0.1:" + wsPort.toString() + "/*"]
                }, (_details, callback) => callback({ cancel: true }));
            }
        });
    });
}
exports.default = startServer;
//# sourceMappingURL=socket.js.map
```

</details>

### Possible solutions

- Move the notices required by MIT to your own License file.
- Do not use WebCord's code at all (I may move parts of it from WebCord and distribute as separate Node modules).

[`socket.ts`]: https://github.com/SpacingBat3/WebCord/blob/master/sources/code/main/modules/socket.ts

**Comment by smartfrigde** [comment]
Resolved in [latest commit](https://github.com/ArmCord/ArmCord/commit/6138281e4c3d58564935efe61604f45699b7cc03)
